### DISCUSSION TITLE
Parts list?

### DISCUSSION THREAD

**Discussion by sqveeze** [chakra-ui/chakra-ui#3682] [Q&A] [answered]
Hello,

Is there anyway to check all part of a component? The documentation describe single part component as:

```
A single part component is a component that returns a single element. For example, the <Button> component renders a <button> HTML element:
```

Well, if i get it right then input should be also a single part component, but you can style input directly in the theme, only if you add parts: ["field"].

I was trying to search for parts list of each component or something but found nothing. It's kinda hard to search for every component parts on the web and in most cases i find nothing.

**Comment by paambaati** [comment, marked as answer]
A shortcut to this is —

1. Find the source file for the component you want to customize. For example, say you want to customize the `Switch` component, you’ll open https://github.com/chakra-ui/chakra-ui/blob/main/packages/switch/src/switch.tsx
2. Find all instances of the string “styles.”. Each of those is a part. In Switch, you’ll find the parts `container`, `track`, `label` and `thumb`.

**Reply by sqveeze** [reply]
In the meantime i already figured it out. However it is not too friendly especially for beginners. I will consider to add a guide to the docs about this in a pr.

**Comment by chr-ge** [comment]
Just look in the component's part array in the default theme: https://github.com/chakra-ui/chakra-ui/tree/main/packages/theme/src/components
